src/statsmodels/contrasts.jl

```diff
@@ -0,0 +1,143 @@
+## Specify contrasts for coding categorical data in model matrix. Contrast types
+## are constructed with data and other options, creating a subtype of
+## AbstractContrast. AbstractContrast provides the interface for creating modle
+## matrix columns and coefficient names
+##
+## ModelFrame will hold a Dict{Symbol, T<:AbstractContrast} that maps column
+## names to contrasts.
+##
+## ModelMatrix will check this dict when evaluating terms, falling back to a
+## default for any categorical data without a specified contrast.
+
+abstract AbstractContrast
+
+termnames(term::Symbol, col::Any, contrast::AbstractContrast) =
+    ["$term - $name" for name in contrast.termnames]
```

Comment:
`"$term $name"` or `"$term.$name"` (or something else) looks like a more common convention to me. `-` could mean "minus" in the context of a regression model.